src/xunit.runner.uap/StandardUapVisitor.cs

```diff
@@ -0,0 +1,190 @@
+        }
+
+        protected override bool Visit(ITestFinished testFinished)
+        {
+            if (showProgress)
+            {
+                log.WriteLine($"   {XmlEscape(testFinished.Test.DisplayName)} [FINISHED] Time: {testFinished.ExecutionTime}s");
+            }
+            return base.Visit(testFinished);
+        }
+
+        protected override bool Visit(IErrorMessage error)
+        {
+            WriteError("FATAL", error);
+
+            return base.Visit(error);
+        }
+
+        protected override bool Visit(ITestAssemblyCleanupFailure cleanupFailure)
+        {
+            WriteError($"Test Assembly Cleanup Failure ({cleanupFailure.TestAssembly.Assembly.AssemblyPath})", cleanupFailure);
+
+            return base.Visit(cleanupFailure);
+        }
+
+        protected override bool Visit(ITestCaseCleanupFailure cleanupFailure)
+        {
+            WriteError($"Test Case Cleanup Failure ({cleanupFailure.TestCase.DisplayName})", cleanupFailure);
+
+            return base.Visit(cleanupFailure);
+        }
+
+        protected override bool Visit(ITestClassCleanupFailure cleanupFailure)
+        {
+            WriteError($"Test Class Cleanup Failure ({cleanupFailure.TestClass.Class.Name})", cleanupFailure);
+
+            return base.Visit(cleanupFailure);
+        }
+
+        protected override bool Visit(ITestCollectionCleanupFailure cleanupFailure)
+        {
+            WriteError($"Test Collection Cleanup Failure ({cleanupFailure.TestCollection.DisplayName})", cleanupFailure);
+
+            return base.Visit(cleanupFailure);
+        }
+
+        protected override bool Visit(ITestCleanupFailure cleanupFailure)
+        {
+            WriteError($"Test Cleanup Failure ({cleanupFailure.Test.DisplayName})", cleanupFailure);
+
+            return base.Visit(cleanupFailure);
+        }
+
+        protected override bool Visit(ITestMethodCleanupFailure cleanupFailure)
+        {
+            WriteError($"Test Method Cleanup Failure ({cleanupFailure.TestMethod.Method.Name})", cleanupFailure);
+
+            return base.Visit(cleanupFailure);
+        }
+
+        protected void WriteError(string failureName, IFailureInformation failureInfo)
+        {
+            log.WriteLine($"   [{failureName}] {XmlEscape(failureInfo.ExceptionTypes[0])}");
+            log.WriteLine($"      {XmlEscape(ExceptionUtility.CombineMessages(failureInfo))}");
+
+            WriteStackTrace(ExceptionUtility.CombineStackTraces(failureInfo));
+        }
+
+        void WriteStackTrace(string stackTrace)
+        {
+            if (String.IsNullOrWhiteSpace(stackTrace))
+                return;
+
+            log.WriteLine("      Stack Trace:");
+
+            foreach (var stackFrame in stackTrace.Split(new[] { Environment.NewLine }, StringSplitOptions.None))
+            {
+                log.WriteLine($"         {StackFrameTransformer.TransformFrame(stackFrame, Directory.GetCurrentDirectory())}");
```

Comment:
>GetCurrentDirectory [](start = 101, length = 19)

does the tests always switch to the test directory as current directory?

Reply:
Yes, I followed the approach of what he have on our netcoreapp xunit runner.